privacyidea/api/lib/utils.py

```diff
@@ -267,3 +268,18 @@ def verify_auth_token(auth_token, required_role=None):
                         "this resource!").format(required_role),
                         id=ERROR.AUTHENTICATE_MISSING_RIGHT)
     return r
+
+
+def check_policy_name(name):
+    """
+    This function checks, if the given name is a valid policy name.
+
+    :param name: The name of the policy
+    :return: Raises a ParameterError in case of an invalid name
+    """
+    if not re.match('^[a-zA-Z0-9_.\- ]*$', name):
+        raise ParameterError(_("The name of the policy may only contain "
+                               "the characters a-zA-Z0-9_.- "))
+
+    if name.lower() == "check":
```

Comment:
Until now, we used dashes in names of migration policies, because this way we were sure that they don't clash with user-defined policies. If we now allow dashes in policy names, we lose this loophole. An idea: Should we disallow to create policies whose names start with ``pi-update-policy-``? This way, we could still create migration policies without worrying about name clashes in the future.

Reply:
We sill can use "~" or "*" or whatever in future migration policies.
Do you really think we need to disallow that? Would of course play into @plettich's hand/suggestion with the disallowed list ;-)

Reply:
Well, I think just disallowing ``pi-update-policy-`` prefixes can be easily implemented now, and it would save us from hassle in the future :) And then we have the advantage that all migration policies have the same prefix.

Reply:
check this out. 
0ea6ecfd

Reply:
thanks!